src/store/ordinoSlice.ts

```diff
@@ -150,18 +141,28 @@ const ordinoSlice = createSlice({
       state.workbenches.splice(action.payload.workbenchIndex);
       state.workbenches.push(action.payload.newWorkbench);
     },
-    addSelection(state, action: PayloadAction<{workbenchIndex: number, viewIndex: number, newSelection: any}>) {
-      // state.workbenches[action.payload.workbenchIndex].views[action.payload.viewIndex].selections = action.payload.newSelection;
+    addSelection(state, action: PayloadAction<{newSelection: number[]}>) {
+      state.workbenches[state.focusViewIndex].selections = action.payload.newSelection;
     },
-    addFilter(state, action: PayloadAction<{workbenchIndex: number, viewIndex: number, newFilter: any}>) {
-      // state.workbenches[action.payload.workbenchIndex].views[action.payload.viewIndex].filters.push(action.payload.newFilter);
+    addFilter(state, action: PayloadAction<{viewId: string, filter: number[]}>) {
+      state.workbenches[state.focusViewIndex].views.find((v) => v.id === action.payload.viewId).filters = action.payload.filter;
     },
     changeFocus(state, action: PayloadAction<{index: number}>) {
       state.focusViewIndex = action.payload.index;
-    }
+    },
+    setWorkbenchData(state, action: PayloadAction<{data: any[]}>) {
+      for(const i of action.payload.data) {
+        state.workbenches[state.focusViewIndex].data[i._id] = i;
+      }
+    },
+    addScoreColumn(state, action: PayloadAction<{columnName: string, data: any}>) {
+      for(const i of action.payload.data) {
```

Comment:
Is `i` a row? If yes, please rename.

```suggestion
      for(const row of action.payload.data) {
```